protocol/chainlib/common.go

```diff
@@ -177,3 +181,10 @@ func verifyTendermintEndpoint(endpoints []string) (websocketEndpoint string, htt
 	}
 	return websocketEndpoint, httpEndpoint
 }
+
+func storeRefererHeaderIfNeeded(c *fiber.Ctx) {
+	isMetricEnabled, _ := strconv.ParseBool(os.Getenv("IS_METRICS_ENABLED"))
```

Comment:
let's not parse this every relay, this is a handler you create one time but every time use. you can create it with the parsing and without it when you don't need it. parse the IS_METRICS_ENABLED outside the constructFiberCallbackWithHeaderAndParameterExtraction
and create the callback accordingly 

will be more efficient that way

Reply:
we should have i think on the MetricsService level, no?

Reply:
we have on portal level ,will put it there also.